versions/3.0.4.md

```diff
@@ -3523,3 +3530,92 @@ Version   | Date       | Notes
 1.2       | 2014-03-14 | Initial release of the formal document.
 1.1       | 2012-08-22 | Release of Swagger 1.1
 1.0       | 2011-08-10 | First release of the Swagger Specification
+
+## <a name="percentEncodingAndFormMediaTypes"></a>Appendix E: Percent-Encoding and Form Media Types
+
+_**NOTE:** In this section, the `application/x-www-form-urlencoded` and `multipart/form-data` media types are abbreviated as `form-urlencoded` and `form-data`, respectively, for readability._
+
+Percent-encoding is used in URIs and media types that derive their syntax from URIs.
+This process is concerned with three sets of characters, the names of which vary among specifications but are defined as follows for the purposes of this section:
+
+* _unreserved_ characters do not need to be percent-encoded; while it is safe to percent-encode them, doing so produces a URI that is [not normalized](https://datatracker.ietf.org/doc/html/rfc3986#section-6.2.2.2)
+* _reserved_ characters either have special behavior in the URI syntax (such as delimiting components) or are reserved for other specifications that need to define special behavior (e.g. `form-urlencoded` defines special behavior for `=`, `&`, and `+`)
+* _unsafe_ characters are known to cause problems when parsing URIs in certain environments
+
+Unless otherwise specified, this section uses RFC3986's definition of [reserved](https://datatracker.ietf.org/doc/html/rfc3986#section-2.2) and [unreserved](https://datatracker.ietf.org/doc/html/rfc3986#section-2.3), and defines the unsafe set as all characters not included in either of those sets.
+
+### Percent-Encoding and `form-urlencoded`
+
+Each URI component (such as the query string) considers some of the reserved characters to be unsafe, either because they serve as delimiters between the components (e.g. `#`), or (in the case of `[` and `]`) were historically considered globally unsafe but were later given reserved status for limited purposes.
+
+Reserved characters with no special meaning defined within a component can be left un-percent encoded.
+However, other specifications can define special meanings, requiring percent-encoding for those characters outside of the additional special meanings.
+
+The `form-urlencoded` media type defines special meanings for `=` and `&` as delimiters, and `+` as the replacement for the space character (instead of its percent-encoded form of `%20`).
+This means that while these three characters are reserved-but-allowed in query strings by RFC3986, the must be percent-encoded in `form-urlencoded` query strings except when used for their `form-urlencoded` purposes; see Appendix C for an example of handling `+` in form values.
```

Comment:
Make "Appendix C" a reference once that appendix has been merged.

Reply:
I have now stubbed out the header and made this (and the other Appendix C reference) link to it.  That seems to be the least objectionable of the three ways I've tried to handle this situation in currently open PRs :-)